client/homescreen/layout.js

```diff
@@ -337,7 +337,7 @@ export default compose(
 			shouldShowWelcomeModal,
 			shouldShowWelcomeFromCalypsoModal,
 			isTaskListHidden: getTaskList( 'setup' )?.isHidden,
-			hasTaskList: taskLists.find( ( list ) => list.isVisible ),
+			hasTaskList: !! taskLists.find( ( list ) => list.isVisible ),
```

Comment:
Unrelated to the bulk of this PR, but I alway notice a warning on the home page that `hasTaskList` expects a boolean, but instead got an object, this fixes that.